### PR TITLE
AST-1315 - Added new design options for dark mode switcher

### PR DESCRIPTION
**Pull request by imnavanath**
### Description
 Design options for dark mode switcher element

### Screenshots
- https://share.bsf.io/rRuWNvzZ

### Types of changes
- New feature - non-breaking change 

### How has this been tested?
- Check every option 
- With light mode
- with dark mode
- Check on frontend
- Check with customizer preview

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Comment by premanshup** [comment]
@imnavanath Please get this synced.

**Comment by premanshup** [comment]
@imnavanath Please check the code analysis too.

File: inc/builder/controllers/class-astra-builder-ui-controller.php [hunk 1]
```diff
@@ -276,24 +276,24 @@ public static function render_site_identity( $device ) {
...
+			$switcher_dark_icon   = ( class_exists( 'Astra_Ext_Extension' ) && Astra_Ext_Extension::is_active( 'dark-mode-switch' ) ) ? astra_get_option( 'mode-switcher-dark-icon' ) : $switcher_light_icon;
 			$switcher_light_label = astra_get_option( 'mode-switcher-light-label' );
 			$switcher_dark_label  = astra_get_option( 'mode-switcher-dark-label' );
 
 			if ( is_customize_preview() ) {
 				self::render_customizer_edit_button();
 			}
 			?>
-				<button class="ast-mode-switcher-trigger" aria-label="Switch to dark mode">
+				<button class="ast-mode-switcher-trigger ast-switcher-<?php echo esc_attr( $switcher_type ); ?>-type ast-switcher-<?php echo esc_attr( $switcher_style ); ?>-style" aria-label="Switch to dark mode">
```
Comment (premanshup) [inline inc/builder/controllers/class-astra-builder-ui-controller.php:290]: @imnavanath Can we reduce the extra CSS classes from here?

File: inc/builder/controllers/class-astra-builder-ui-controller.php [hunk 2]
```diff
@@ -276,24 +276,24 @@ public static function render_site_identity( $device ) {
...
 					<?php
-					switch ( $switcher_style ) {
+					switch ( $switcher_type ) {
 						case 'icon':
-							echo '<span class="ast-light-mode-wrap">' . self::fetch_svg_icon( $switcher_light_icon ) . esc_html( $switcher_light_label ) . '</span>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+							echo '<span class="ast-light-mode-wrap ast-mode-label" data-tooltip="' . esc_html( 'Light' ) . '">' . self::fetch_svg_icon( $switcher_light_icon ) . '</span>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 
-							echo '<span class="ast-dark-mode-wrap">' . self::fetch_svg_icon( $switcher_dark_icon ) . esc_html( $switcher_dark_label ) . '</span>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+							echo '<span class="ast-dark-mode-wrap ast-mode-label" data-tooltip="' . esc_html( 'Dark' ) . '">' . self::fetch_svg_icon( $switcher_dark_icon ) . '</span>'; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```
Comment (premanshup) [inline inc/builder/controllers/class-astra-builder-ui-controller.php:296]: @imnavanath Can we remove extra CSS classes from here?

File: inc/customizer/configurations/builder/header/class-astra-header-mode-switcher-component-configs.php [hunk 3]
```diff
@@ -59,11 +59,11 @@ public function register_configuration( $configurations, $wp_customize ) {
 				 * Option: Color palette selection.
 				 */
 				array(
-					'name'      => ASTRA_THEME_SETTINGS . '[dark-mode-switch-style]',
-					'default'   => astra_get_option( 'dark-mode-switch-style' ),
+					'name'      => ASTRA_THEME_SETTINGS . '[dark-mode-switch-type]',
+					'default'   => astra_get_option( 'dark-mode-switch-type' ),
```
Comment (premanshup) [inline inc/customizer/configurations/builder/header/class-astra-header-mode-switcher-component-configs.php:63]: @imnavanath Any reason we are changing the name of this option? 
`dark-mode-switch-style` this option is still used in other places as well. Please verify once.

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/builder/controllers/class-astra-builder-ui-controller.php (hunk 2 above)
Comment (imnavanath) [reply inc/builder/controllers/class-astra-builder-ui-controller.php:297]: Yes we can remove that, actually I made a common `ast-mode-label` class for both light & dark span, to apply common stylings.

File: inc/customizer/configurations/builder/header/class-astra-header-mode-switcher-component-configs.php (hunk 3 above)
Comment (imnavanath) [reply inc/customizer/configurations/builder/header/class-astra-header-mode-switcher-component-configs.php]: Yes intentionally it was changed. 
TYPE means switcher type should br of icon/label/icon-width-label.
STYLE means how it should display toggle/button format. 
Previously I worked with type then I realized we are going to add new control for style as well that's why I renamed this accordingly.

File: inc/builder/controllers/class-astra-builder-ui-controller.php (hunk 1 above)
Comment (imnavanath) [reply inc/builder/controllers/class-astra-builder-ui-controller.php]: Compatibility CSS I written on basis of these class like ast-switcher-icon/ast-switcher-label/ast-switcher-icon-with-label.
BTW, I removed the class `ast-switcher-toggle/button` and loaded only when pro & fixed-switch addon is active. https://share.bsf.io/5zu92kXd

File: inc/core/builder/class-astra-builder-options.php [hunk 4]
```diff
@@ -1035,11 +1035,15 @@ function astra_hf_builder_customizer_defaults( $defaults ) {
...
 	 * Mode Switcher element.
 	*/
+	$defaults['dark-mode-switch-type']          = 'icon';
 	$defaults['mode-switcher-light-icon']       = 'light-switcher-1';
-	$defaults['mode-switcher-light-label']      = '';
-	$defaults['mode-switcher-dark-label']       = '';
+	$defaults['mode-switcher-light-label']      = __( 'Light', 'astra' );
+	$defaults['mode-switcher-dark-label']       = __( 'Dark', 'astra' );
+	$defaults['dark-mode-switcher-light-color'] = '#0170b9';
+	$defaults['dark-mode-switcher-dark-color']  = '#eeeeee';
```
Comment (premanshup) [inline inc/core/builder/class-astra-builder-options.php:1043]: @imnavanath Please check if can integrate this with Global Colors Pallete?

File: inc/markup-extras.php [hunk 5]
```diff
@@ -187,9 +187,9 @@ function astra_logo( $device = 'desktop', $echo = true ) {
 		$site_title           = astra_get_option( 'display-site-title-responsive' );
 		$display_site_title   = ( $site_title['desktop'] || $site_title['tablet'] || $site_title['mobile'] ) ? true : false;
 
-		$html = '';
-
+		$html            = '';
 		$has_custom_logo = apply_filters( 'astra_has_custom_logo', has_custom_logo() );
+		do_action( 'astra_before_logo_image', $device );
```
Comment (premanshup) [inline inc/markup-extras.php:192]:
```suggestion
		do_action( 'astra_before_logo_image', $device );
```
@imnavanath Any reason we are introducing this action?

Comment (imnavanath) [reply inc/markup-extras.php:192]: We added this new action to insert dark mode logo which will come from addon, you can ref that here - https://github.com/brainstormforce/astra-addon/blob/pro-dark-mode-features/addons/dark-mode-switch/classes/class-astra-addon-dark-mode-markup.php#L68

File: assets/js/unminified/mode-switcher.js [hunk 6]
```diff
@@ -19,6 +19,27 @@ updateSwitcherAtts = function ( switchedTo ) {
 	}
 }
 
+// Update mode switcher flash message.
+updateSwitcherFlashMessages = function ( switchedTo ) {
```
[the remark this answers is not in the document]
Comment (imnavanath) [reply assets/js/unminified/mode-switcher.js]: As flash message removed this code is no longer exists.

File: inc/core/builder/class-astra-builder-options.php (hunk 4 above)
Comment (imnavanath) [reply inc/core/builder/class-astra-builder-options.php]: Done = https://github.com/brainstormforce/astra/pull/3706/commits/746b7bd8023c742808b00a07741feb1a7fc8c830

**Pull request merged**